UI/FleetButton.cpp

```diff
@@ -291,6 +313,40 @@ void FleetButton::LayoutIcons() {
         GG::Pt graphic_ul = middle - subtexture_sz / 2;
         m_selection_indicator->SizeMove(graphic_ul, graphic_ul + subtexture_sz);
     }
+
+    // refresh fleet button tooltip
+    if (m_fleet_blockaded) {
+        std::shared_ptr<Fleet> fleet;
+        std::shared_ptr<System> current_system;
+        std::map<int, bool> adjacent_system_IDs;
+        std::string available_exits = "";
+        int available_exits_count = 0;
+
+        if (!m_fleets.empty())
+            // can just pick first fleet because all fleets in system should have same exits
+            fleet = GetFleet(*m_fleets.begin());
```

Comment:
On GG::Rotating/StaticGraphic: as there doesn't seem to be a good reason why the fb shouldn't be a StaticGraphic, I will change it as Geoff and you suggested.